### PR TITLE
[TypeDeclaration] Add AddParamTypeSplFixedArrayRector

### PR DESCRIPTION
**Pull request by TomasVotruba**

### CODE REVIEW

File: rules/TypeDeclaration/Rector/FunctionLike/AddParamTypeSplFixedArrayRector.php [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+class SomeClass
+{
+    public function run(Tokens $tokens)
+    {
+    }
+}
+CODE_SAMPLE
+                    ,
+                    <<<'CODE_SAMPLE'
+use PhpCsFixer\Tokenizer\Token;
```
Comment (samsonasik) [inline rules/TypeDeclaration/Rector/FunctionLike/AddParamTypeSplFixedArrayRector.php:64]: the use add is only on `importNames()` enabled, can be removed: 

```suggestion
```

File: rules/TypeDeclaration/Rector/FunctionLike/AddParamTypeSplFixedArrayRector.php [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+use PhpCsFixer\Tokenizer\Tokens;
+
+class SomeClass
+{
+    /**
+     * @param Tokens<Token>
```
Comment (samsonasik) [inline rules/TypeDeclaration/Rector/FunctionLike/AddParamTypeSplFixedArrayRector.php:70]: Fqcn: 

```suggestion
     * @param \PhpCsFixer\Tokenizer\Tokens<\PhpCsFixer\Tokenizer\Token> $tokens
```

File: rules/TypeDeclaration/Rector/FunctionLike/AddParamTypeSplFixedArrayRector.php [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+                continue;
+            }
+
+            $paramName = $this->getName($param);
+            $this->phpDocTypeChanger->changeParamType($functionLikePhpDocInfo, $genericParamType, $param, $paramName);
+
+            $hasChanged = true;
+        }
+
+        if ($hasChanged) {
```
Comment (samsonasik) [inline rules/TypeDeclaration/Rector/FunctionLike/AddParamTypeSplFixedArrayRector.php]: `$functionLikePhpDocInfo->hasChanged()` should be the way to verify

File: rules/TypeDeclaration/Rector/FunctionLike/AddParamTypeSplFixedArrayRector.php [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+            return $node;
+        }
+
+        return null;
+    }
+
+    private function resolveGenericType(TypeWithClassName $typeWithClassName): GenericObjectType|null
+    {
+        foreach (self::SPL_FIXED_ARRAY_TO_SINGLE as $fixedArrayClass => $singleClass) {
+            if ($typeWithClassName->getClassName() === $fixedArrayClass) {
```
Comment (samsonasik) [inline rules/TypeDeclaration/Rector/FunctionLike/AddParamTypeSplFixedArrayRector.php:130]: repetitive usage `$typeWithClassName->getClassName()` can be moved to.variable before loop

File: rules/TypeDeclaration/Rector/FunctionLike/AddParamTypeSplFixedArrayRector.php [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+    public function refactor(Node $node): ?Node
+    {
+        if ($node->getParams() === []) {
+            return null;
+        }
+
+        $functionLikePhpDocInfo = $this->phpDocInfoFactory->createFromNodeOrEmpty($node);
+        $hasChanged = false;
+
+        foreach ($node->getParams() as $param) {
```
Comment (samsonasik) [inline rules/TypeDeclaration/Rector/FunctionLike/AddParamTypeSplFixedArrayRector.php:93]: Repetitive `$node->getParams()` can be moved to variable, or just use `$node->params` instead

Comment (TomasVotruba) [reply rules/TypeDeclaration/Rector/FunctionLike/AddParamTypeSplFixedArrayRector.php:93]: This method is only static getter. It depends on `FunctionLike` interface, so the `params` property is not available there :disappointed:

File: rules/TypeDeclaration/Rector/FunctionLike/AddParamTypeSplFixedArrayRector.php (hunk 4 above)
Comment (TomasVotruba) [reply rules/TypeDeclaration/Rector/FunctionLike/AddParamTypeSplFixedArrayRector.php:130]: This is just getter with static value, so it's ok here

File: rules/TypeDeclaration/Rector/FunctionLike/AddParamTypeSplFixedArrayRector.php (hunk 3 above)
Comment (TomasVotruba) [reply rules/TypeDeclaration/Rector/FunctionLike/AddParamTypeSplFixedArrayRector.php]: Thanks :+1:

File: rules/TypeDeclaration/Rector/FunctionLike/AddParamTypeSplFixedArrayRector.php (hunk 4 above)
Comment (samsonasik) [reply rules/TypeDeclaration/Rector/FunctionLike/AddParamTypeSplFixedArrayRector.php:130]: Ok 👍

File: rules/TypeDeclaration/Rector/FunctionLike/AddParamTypeSplFixedArrayRector.php (hunk 5 above)
Comment (samsonasik) [reply rules/TypeDeclaration/Rector/FunctionLike/AddParamTypeSplFixedArrayRector.php:93]: Ok 👍

File: rules/TypeDeclaration/Rector/FunctionLike/AddParamTypeSplFixedArrayRector.php (hunk 2 above)
Comment (TomasVotruba) [reply rules/TypeDeclaration/Rector/FunctionLike/AddParamTypeSplFixedArrayRector.php:70]: Examples have convention to use the imported names, as easier to read and convention.

**Comment by TomasVotruba** [comment]
Thanks for feedback @samsonasik , I've applied few changes

**Pull request merged**